components/Faucet.tsx

```diff
@@ -7,29 +7,30 @@ import FaucetButton from "@/components/FaucetButton";
 import FaucetAccountInput from "@/components/FaucetAccountInput";
 import TokenPicker from "@/components/TokenPicker";
 import { supportedTokens } from "@/libs/utils/supportedTokens";
-import { CENNZnetToken } from "@/libs/types";
+import { CENNZnetToken, TxStatus } from "@/libs/types";
+import FaucetProgress from "@/components/FaucetProgress";
 
 const Faucet: FC = () => {
 	const [token, setToken] = useState<CENNZnetToken>(supportedTokens[0]);
 	const [network, setNetwork] = useState<string>(NETWORKS[0]);
 	const [address, setAddress] = useState<string>("");
-	const [response, setResponse] = useState<string>();
-	const [fetchingResponse, setFetchingResponse] = useState<boolean>(false);
+	const [isOpen, setIsOpen] = useState<boolean>(false);
+	const [response, setResponse] = useState<TxStatus>();
 
 	const fetchSupplyResponse = async () => {
 		if (!address || !network) return;
-
-		setFetchingResponse(true);
+		setResponse({
+			message: "Retrieving Tokens from the Faucet",
+			status: "in-progress",
+		});
```

Comment:
could you specify which token is being sent here?

Reply:
oo yeah good call